kingpin/actors/aws/elb.py

```diff
@@ -365,6 +365,150 @@ def _execute(self):
+
+    **Dry run**
+
+    Will check that Cert names are existent, and will also check that the
+    credentials provided for AWS have access to the new cert for ssl.
+    """
+
+    all_options = {
+        'region': (str, REQUIRED, 'AWS region (or zone) name, like us-west-2'),
+        'old_cert': (str, REQUIRED, 'Unique IAM certificate name, or ARN'),
+        'new_cert': (str, REQUIRED, 'Unique IAM certificate name, or ARN'),
+    }
+
+    @gen.coroutine
+    def _check_access(self, elb):
+        """Perform a dummy operation to check credential accesss.
+
+        Intended to be used in a dry run, this method attempts to perform an
+        invalid set_listener call and monitors the output of the error. If the
+        error is anything other than AccessDenied then the provided credentials
+        are sufficient and we do nothing.
+
+        Args:
+            elb: boto LoadBalancer object.
+        """
+        try:
+            # A blank ARN value should have code 'CertificateNotFound'
+            # We're only checking if credentials have sufficient access
+            yield self.thread(
+                elb.set_listener_SSL_certificate,
+                self.option('port'),
+                '')
+        except BotoServerError as e:
+            if e.error_code == 'AccessDenied':
+                raise exceptions.InvalidCredentials(e)
+
+    @gen.coroutine
+    def _get_cert_arn(self, name):
+        """Return a server_certificate ARN.
+
+        Searches for a certificate object and returns the "ARN" value.
+
+        Args:
+            name: certificate name
+
+        Raises:
+            CertNotFound - if the name doesn't match an existing cert.
+
+        Returns:
+            string: the ARN value of the certificate
+        """
+
+        self.log.debug('Searching for cert "%s"...' % name)
+        try:
+            cert = yield self.thread(
+                self.iam_conn.get_server_certificate, name)
+        except BotoServerError as e:
+            raise CertNotFound(
+                'Could not find cert %s. Reason: %s' % (name, e))
+
+        # Get the ARN of this cert
+        arn = cert['get_server_certificate_response'].get(
+            'get_server_certificate_result').get(
+            'server_certificate').get(
+            'server_certificate_metadata').get('arn')
+
+        raise gen.Return(arn)
+
+    @gen.coroutine
+    def _use_cert(self, elb, arn, port):
+        """Assign an ssl cert to a given ELB.
+
+        Args:
+            elb: boto elb object.
+            arn: ARN for server certificate to use.
+        """
+
+        self.log.info('Setting ELB "%s" to use cert arn: %s' % (elb, arn))
+        try:
+            yield self.thread(
+                elb.set_listener_SSL_certificate, port, arn)
+        except BotoServerError as e:
+            raise exceptions.RecoverableActorFailure(
+                'Applying new SSL cert to %s failed: %s' % (elb, e))
+
+    def _find_cert_port(self, elb, new_arn):
```

Comment:
Little formatting and doc string issues here.. remove the extra line, document the code.